tests/TestCli.cpp

```diff
@@ -250,106 +259,80 @@ void TestCli::testAdd()
     QVERIFY(!addCmd.name.isEmpty());
     QVERIFY(addCmd.getDescriptionLine().contains(addCmd.name));
 
-    Utils::Test::setNextPassword("a");
-    addCmd.execute({"add",
-                    "-u",
-                    "newuser",
-                    "--url",
-                    "https://example.com/",
-                    "-g",
-                    "-L",
-                    "20",
-                    m_dbFile->fileName(),
-                    "/newuser-entry"});
-    m_stderrFile->reset();
-    m_stdoutFile->reset();
-    m_stdoutFile->readLine(); // skip password prompt
-    QCOMPARE(m_stderrFile->readAll(), QByteArray(""));
-    QCOMPARE(m_stdoutFile->readAll(), QByteArray("Successfully added entry newuser-entry.\n"));
-
-    auto db = readTestDatabase();
+    setInput("a");
+    execCmd(addCmd,
+            {"add",
+             "-u",
+             "newuser",
+             "--url",
+             "https://example.com/",
+             "-g",
+             "-L",
+             "20",
+             m_dbFile->fileName(),
+             "/newuser-entry"});
+    QVERIFY(m_stderr->readAll().contains("Enter password to unlock"));
+    QVERIFY(m_stdout->readAll().contains("Successfully added entry newuser-entry."));
```

Comment:
This version of the assertion is a bit more lax than it was before. Previously we were testing that the output is exactly this line. I would preserve the original version if possible.

Reply:
Same goes for the assertion just above, where we could test the first line only using `read` instead of `readAll`. There's a few more assertions that could be stricter in the other test cases.

Reply:
@louib I want to push back on this a little. The reason I moved to readall was to reduce the amount of churn going into these tests. The **purpose** of the test is to determine the ultimate success or failure of a particular action. We are not testing that the entire sequence of verbiage is exactly what was programmed in.

In that context, I think it makes sense to leave these "relaxed" as long as they are testing the expected **outcome** of a CLI command.

Reply:
Fair point. But in the context of a CLI the textual output of the program, whether it's using `stderr` or `stdout`, is also part of the API, so I think it makes sense to also test what was written on various streams by the program. I'm not saying we should do that systematically but I don't see a reason why to remove the checks that we have. In my experience, updating those assertions was never the bottleneck when adding features to the CLI, with the exception of changes like in this PR.

I don't have a hard opinion on this and this PR is great work anyway. I don't think this should be blocking the PR from being merged.

Reply:
I will add the stderr checks in after the first command to confirm only the password prompt was shown to the user. Thanks!!